modules/apps/dynamic-data-mapping/dynamic-data-mapping-api/src/main/java/com/liferay/dynamic/data/mapping/storage/constants/FieldConstants.java

```diff
@@ -187,15 +188,20 @@ public static Serializable getSerializable(
 		else if (type.equals(FieldConstants.DATE)) {
 			return values.toArray(new String[0]);
 		}
-		else if (type.equals(FieldConstants.DOUBLE)) {
-			return values.toArray(new Number[0]);
+		else if (type.equals(FieldConstants.DOUBLE) ||
+				 type.equals(FieldConstants.INTEGER)) {
+
+			List<Number> numberValues = new ArrayList<>();
+
+			for (Serializable value : values) {
+				numberValues.add(GetterUtil.getNumber(value));
+			}
+
+			return numberValues.toArray(new Number[0]);
```

Comment:
This logic already exists in Liferay. Check this out:
```suggestion
			return ListUtil.toArray(
				values,
				new Accessor<Object, Number>() {

					@Override
					public Number get(Object value) {
						return GetterUtil.getNumber(value);
					}

					@Override
					public Class<Number> getAttributeClass() {
						return Number.class;
					}

					@Override
					public Class<Object> getTypeClass() {
						return Object.class;
					}

				});
```

Reply:
hi @rodrigopaulino thanks for the suggestion! I have applied and force-pushed it to this pull.